examples/basic.rs

```diff
@@ -0,0 +1,128 @@
+extern crate futures;
+extern crate futures_state_stream;
+extern crate tokio_core;
+extern crate tokio_imap;
+
+use futures::future::Future;
+use futures_state_stream::StateStream;
+use std::error::Error;
+use std::io;
+use std::fmt::{self, Display, Formatter};
+use tokio_core::reactor::Core;
+use tokio_imap::client::Client;
+use tokio_imap::client::builder::{CommandBuilder, FetchBuilderAttributes, FetchBuilderMessages,
+                                  FetchBuilderModifiers};
+use tokio_imap::proto::{Attribute, AttributeValue, Response, ResponseData};
+
+fn process_email(responsedata: &ResponseData) {
+    if let Response::Fetch(_, ref attr_vals) = *responsedata.parsed() {
+        for val in attr_vals.iter() {
+            match *val {
+                AttributeValue::Uid(u) => {
+                    eprintln!("E-mail UID: {}", u);
+                }
+                AttributeValue::Rfc822(Some(src)) => {
+                    eprintln!("E-mail body length: {}", src.to_vec().len());
+                }
+                _ => (),
+            }
+        }
+    }
+}
+
+#[derive(Debug)]
+pub enum SyncError {
```

Comment:
Since all of these errors are of the same time, I don't think having the `SyncError` type in this example makes a lot of sense.

Reply:
Of the same type you mean? The variants allow pinpointing what error occurred specifically.

Reply:
Okay, I guess that makes sense. In this case it would be more idiomatic to have a tuple variants rather than struct variants, though.